### PR TITLE
Tweak/experiments dom optimization migration - ED-597

### PR DESCRIPTION
**Pull request by kobizz**

### CODE REVIEW

File: core/upgrade/upgrades.php [hunk 1]
```diff
@@ -863,6 +864,18 @@ public static function _v_3_0_0_move_default_typography_to_kit( $updater, $inclu
...
 
+	public static function v_3_1_0_move_optimized_dom_output_to_experiments() {
+		$saved_option = get_option( 'elementor_optimized_dom_output' );
+
+		if ( $saved_option ) {
+			$new_option = 'enabled' === $saved_option ? Experiments_Manager::STATE_ACTIVE : Experiments_Manager::STATE_INACTIVE;
+
+			add_option( 'elementor_experiment-e_dom_optimization', $new_option );
+
+			delete_option( 'elementor_optimized_dom_output' );
```
Comment (matipojo) [inline core/upgrade/upgrades.php:875]: Don't delete the option, it should still work after a downgrade.

File: includes/elements/section.php [hunk 2]
```diff
@@ -434,8 +434,7 @@ protected function _register_controls() {
 			]
 		);
 
-		$is_legacy_mode_active = Plugin::instance()->get_legacy_mode( 'elementWrappers' );
-		$content_position_selector = $is_legacy_mode_active ? '{{WRAPPER}} > .elementor-container > .elementor-row > .elementor-column > .elementor-column-wrap > .elementor-widget-wrap' : '{{WRAPPER}} > .elementor-container > .elementor-column > .elementor-widget-wrap';
+		$content_position_selector = Plugin::$instance->experiments->is_feature_active( 'e_dom_optimization' ) ? '{{WRAPPER}} > .elementor-container > .elementor-column > .elementor-widget-wrap' : '{{WRAPPER}} > .elementor-container > .elementor-row > .elementor-column > .elementor-column-wrap > .elementor-widget-wrap';
```
Comment (matipojo) [inline includes/elements/section.php:437]: Make it more readable.
```suggestion
		$content_position_selector = Plugin::$instance->experiments->is_feature_active( 'e_dom_optimization' )
		? 
		'{{WRAPPER}} > .elementor-container > .elementor-column > .elementor-widget-wrap' 
		:
		 '{{WRAPPER}} > .elementor-container > .elementor-row > .elementor-column > .elementor-column-wrap > .elementor-widget-wrap';
```

File: tests/phpunit/elementor/includes/settings/test-settings.php [hunk 3]
```diff
@@ -11,7 +11,6 @@ public function test_clear_css_cache_on_update_css_settings() {
 			'elementor_disable_color_schemes',
 			'elementor_disable_typography_schemes',
 			'elementor_css_print_method',
-			'elementor_optimized_dom_output',
```
Comment (matipojo) [inline tests/phpunit/elementor/includes/settings/test-settings.php:14]: Replace with new option name.

File: core/editor/editor.php [hunk 4]
```diff
@@ -588,10 +592,7 @@ public function enqueue_scripts() {
 				'darkModeStylesheetURL' => ELEMENTOR_ASSETS_URL . 'css/editor-dark-mode' . $suffix . '.css',
 				'defaultGenericFonts' => $kits_manager->get_current_settings( 'default_generic_fonts' ),
 			],
-			// Legacy Mode - for backwards compatibility of older HTML markup.
```
Comment (matipojo) [inline core/editor/editor.php:591]: Where is the BC?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/editor/editor.php (hunk 4 above)
Comment (kobizz) [reply core/editor/editor.php]: https://github.com/elementor/elementor/blob/release/3.1.0/assets/dev/js/editor/editor-base.js#L789

File: tests/phpunit/elementor/includes/settings/test-settings.php (hunk 3 above)
Comment (kobizz) [reply tests/phpunit/elementor/includes/settings/test-settings.php]: It's not related to the Settings class anymore

**Review by matipojo — APPROVED** [review]

File: tests/phpunit/elementor/includes/settings/test-settings.php (hunk 3 above)
Comment (matipojo) [reply tests/phpunit/elementor/includes/settings/test-settings.php]: Where you test if it clears the cache on cahnge?

**Pull request merged**